nss-day-cohort-45/tabloidfullstack-giving-gillyflowers
# Koh add tag to comment

First off... I named this branch wrong... (and only realized it on Saturday). This branch is to add tags to POSTS... not comments. Ope.

## Changes
1. Updated methods to delete all postTags associated with a post and then create new entries when a user adds or removes tags from their post.
2. Ternary to display "Manage Tags" button if a user is viewing their own post.
3. A PostTag checkbox form to let users select and unselect tags to go with their posts.
4. When viewing PostDetails, user will see a list of tags associated with the Post. If there are no tags associated with the post, they'll see a no tags message instead.
5. If a tag is deleted in Tag Management, it'll remove the instances of the tag if it's listed on a post.

## To Test
1. Fetch kohAddTagToComment
2. Serve locally and login.
3. Navigate to a post details page, and try adding tags to a post for the logged in user.
4. Trying adding different tags to the same post. 
5. Verify the postTags table in the DB is updating.
6. Go to a post that is not written by the current user. (You should not see the Manage Tags button.)
7. Go to Tag Management and delete one of the tags you've added to a post. Go check that it's no longer on that post.

Of Note: The Url is not protected - if you manually enter it, you can add tags to anyone's post at the moment.
